core/UpdateCheck.php

```diff
@@ -51,6 +52,13 @@ public static function check($force = false, $interval = null)
                 $latestVersion = '';
             }
 
+            $hasLastCheckFailed = '' === $latestVersion;
+            if ($hasLastCheckFailed) {
+                // retry check on next request if previous attempt failed
+                Option::set(self::LAST_TIME_CHECKED, $lastTimeChecked, $autoLoad = 1);
+            }
+
+            Option::set(self::LAST_CHECK_FAILED, $hasLastCheckFailed, $autoLoad = 1);
             Option::set(self::LATEST_VERSION, $latestVersion);
```

Comment:
Good point, it should not be worth setting the version if the check failed.

While it would prevent having "not the most up-to-date version" displayed it would also hide a previously available update if the request failed.